modules/remotebackend/httpconnector.cc

```diff
@@ -435,15 +401,11 @@ int HTTPConnector::recv_message(rapidjson::Document &output) {
       return -1;
     }
 
-    rapidjson::StringStream ss(resp.body.c_str());
     int rv = -1;
-    output.ParseStream<0>(ss);
-
-    // offer whatever we read in send_message
-    if (output.HasParseError() == false)
-       rv = rd;
-    else
-       rv = -1;
+    std::string err;
+    output = Json::parse(resp.body, err);
+    if (output != nullptr) return resp.body.size();
+    L<<Logger::Error<<"Cannot parse JSON reply: "<<err<<endl;
```

Comment:
as long as resp.body is a std::string, output can never be nullptr, but it might be Json::null! (so check with is_null())